### PR TITLE
fix check for systemd-timesyncd.service

### PR DESCRIPTION
**Pull request by jepio**
systemd-timesyncd.service will not be the only ntp implementation enabled by default on all platforms, so the test needs to take that into account.

### CODE REVIEW

File: kola/tests/coretest/core.go [hunk 1]
```diff
@@ -322,26 +322,41 @@ func TestServicesActive() error {
...
-	for _, unit := range units {
+	for _, unit := range allOf {
 		c := exec.Command("systemctl", "is-active", unit)
 		err := c.Run()
 		if err != nil {
-			return fmt.Errorf("Services Active: %v", err)
+			return fmt.Errorf("services Active: %s: %v", unit, err)
+		}
+	}
+	anyActive := false
```
Comment (pothos) [inline kola/tests/coretest/core.go:348]:
```suggestion
	anyActive := false
	var err error
```
How about defining the error here to be able to report it below?

File: kola/tests/coretest/core.go [hunk 2]
```diff
@@ -322,26 +322,41 @@ func TestServicesActive() error {
...
+	for _, unit := range anyOf {
+		c := exec.Command("systemctl", "is-active", unit)
+		err := c.Run()
```
Comment (pothos) [inline kola/tests/coretest/core.go:351]:
```suggestion
		// Note: Don't shadow the error here, i.e., don't use err :=
		err = c.Run()
```

Comment (jepio) [reply kola/tests/coretest/core.go:351]: we expect some of these to fail :) the ntp services can't all be running at the same time

File: kola/tests/coretest/core.go [hunk 3]
```diff
@@ -322,26 +322,41 @@ func TestServicesActive() error {
...
+		if err == nil {
+			anyActive = true
+			break
 		}
 	}
+	if !anyActive {
+		return fmt.Errorf("no NTP service active")
```
Comment (pothos) [inline kola/tests/coretest/core.go:358]:
```suggestion
		return fmt.Errorf("no NTP service active (error: %v)", err)
```
That also works with a nil error shown as `<nil>` but I think we wouldn't run into this

File: kola/tests/coretest/core.go (hunk 2 above)
Comment (jepio) [reply kola/tests/coretest/core.go]: alright

File: kola/tests/coretest/core.go [hunk 4]
```diff
@@ -322,26 +322,41 @@ func TestServicesActive() error {
...
+			return fmt.Errorf("services Active: %s: %v", unit, err)
+		}
+	}
+	anyActive := false
+	for _, unit := range anyOf {
+		c := exec.Command("systemctl", "is-active", unit)
+		err := c.Run()
+		if err == nil {
+			anyActive = true
+			break
```
[the remark this answers is not in the document]
Comment (jepio) [reply kola/tests/coretest/core.go]: ok

**Review by pothos — APPROVED** [review]

**Comment by jepio** [comment]
Test ran successfully. http://jenkins.infra.kinvolk.io:8080/job/container/job/test/20926/

**Pull request merged**